src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
+        # Assign partners of reference person as partners
+        partners_of_rp_parent_ids = household_structure.loc[
+            children_of_rp_idx.intersection(partners_idx)
+        ]
+        if len(partners_of_rp_parent_ids) > 0:
+            partners_of_rp_parent_ids = (
+                partners_of_rp_parent_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[partners_of_rp_parent_ids.index, "guardian_2"] = partners_of_rp_parent_ids
+
+        # Children are relative of reference person with relative(s) in age bound - orange box
+        relatives_with_guardian = household_structure.loc[
+            child_relative_of_rp_idx.intersection(relatives_of_rp_idx).intersection(
+                age_bound_idx
+            )
+        ]
+        if len(relatives_with_guardian) > 0:
+            relatives_with_guardian_ids = (
+                relatives_with_guardian.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                relatives_with_guardian_ids.index, "guardian_1"
+            ] = relatives_with_guardian_ids
+
+        # Children are relative of reference person with no age bound relative(s) - yellow box
+        relative_ids = (
+            household_structure.loc[
+                child_relative_of_rp_idx.drop(
+                    relatives_with_guardian.index.unique("child_id"), level="child_id"
+                ).intersection(rp_idx)
+            ]
+            .reset_index()
+            .set_index("child_id")["person_id"]
+        )
+        pop.loc[relative_ids.index, "guardian_1"] = relative_ids
+        # Assign guardian to spouse/partner of reference person
+        relative_with_rp_partner_guardian_ids = household_structure.loc[
+            child_relative_of_rp_idx.drop(
+                relatives_with_guardian.index.unique("child_id"), level="child_id"
+            ).intersection(partners_idx)
+        ]
+        if len(relative_with_rp_partner_guardian_ids) > 0:
+            relative_with_rp_partner_guardian_ids = (
+                relative_with_rp_partner_guardian_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                relative_with_rp_partner_guardian_ids.index, "guardian_2"
+            ] = relative_with_rp_partner_guardian_ids
+
+        # Children are reference person and have a parent in household - green box
+        child_rp_with_parent = household_structure.loc[
+            child_rp_idx.intersection(parents_of_rp_idx)
+        ]
+        if len(child_rp_with_parent) > 0:
+            child_rp_with_parent_ids = (
+                child_rp_with_parent.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
```

Comment:
I thought we were supposed to assign a random two parents if there are multiple?

Reply:
Syl confirmed to keep it at 1 for now.  There was confusion between the documentation and what was stated in a slack thread.